src/react/databrowser/objects/ObjectDelete.tsx

```diff
@@ -130,6 +129,170 @@ function ConfirmationInput({ toggledFiles }: { toggledFiles: unknown[] }) {
   );
 }
 
+export const enum WarningTypes {
+  LEGAL_HOLD,
+  COMPLIANCE,
+  GOVERNANCE,
+}
+
+function getWarningBanner(warningType: WarningTypes) {
+  if (warningType === WarningTypes.GOVERNANCE) {
+    return (
+      <Box marginTop={spacing.sp8} marginBottom={spacing.sp8}>
+        <Banner icon={<Icon name="Exclamation-circle" />} variant="warning">
+          At least one object you want to delete is under Object-Lock retention{' '}
+          <Icon name="Lock" color="buttonSecondary" /> with governance mode.
+        </Banner>
+        <Box marginTop={spacing.sp8}>
+          <div>
+            Protected versions won't be deleted unless you choose to bypass
+          </div>
+          the governance retention.
+        </Box>
+      </Box>
+    );
+  }
+  if (warningType === WarningTypes.COMPLIANCE) {
+    return (
+      <Box marginTop={spacing.sp8} marginBottom={spacing.sp8}>
+        <Banner icon={<Icon name="Exclamation-triangle" />} variant="warning">
+          At least one object you want to delete is under Object-Lock retention{' '}
+          <Icon name="Lock" color="buttonSecondary" /> with compliance mode.{' '}
+          <br />
+          Object versions under compliance retention cannot be deleted.
+        </Banner>
+      </Box>
+    );
+  }
+
+  if (warningType === WarningTypes.LEGAL_HOLD) {
+    return (
+      <Box marginTop={spacing.sp8} marginBottom={spacing.sp8}>
+        <Banner icon={<Icon name="Exclamation-triangle" />} variant="warning">
+          At least one object you want to delete is under Legal Hold{' '}
+          <Icon name="Lock" color="buttonSecondary" /> <br />
+          Object versions with Legal Hold cannot be deleted.
+        </Banner>
+      </Box>
+    );
+  }
+
+  return <></>;
+}
+
+export const getMessagesAndRequiredActions = ({
+  numberOfObjects,
+  selectedObjectsAreSpecificVersions,
+  isBucketVersioned,
+  objectsLockedInComplianceModeLength,
+  objectsLockedInGovernanceModeLength,
+  objectsLockedInLegalHoldLength,
+}: {
+  numberOfObjects: number;
+  selectedObjectsAreSpecificVersions: boolean;
+  isBucketVersioned: boolean;
+  objectsLockedInComplianceModeLength: number;
+  objectsLockedInGovernanceModeLength: number;
+  objectsLockedInLegalHoldLength: number;
+}) => {
```

Comment:
nit: I would explicitly type the return type of this function to enforce coherence between the numerous branches.

```suggestion
}): {
  info: string,
  warnings: WarningTypes[],
  checkboxRequired: boolean,
  confirmationRequired: boolean,
  isDeletionPossible: boolean
} => {
```